src/app/Trade.ts

```diff
@@ -38,47 +38,81 @@ export class Trade {
   }
 
   private areOffersInvalid(
-    [animal, count]: Offer,
+    offer: Offer[],
     playerHerd: Herd,
-    [bankAnimal, bankCount]: Offer,
+    target: Offer[],
   ): boolean {
     return (
-      (count > 1 && bankCount > 1) ||
-      playerHerd.getAnimalNumber(animal) < count ||
-      this.bank.theHerd.getAnimalNumber(bankAnimal) < bankCount
+      (offer.length > 1 && target.length > 1) ||
+      this.isOfferQuantityInvalid(offer, playerHerd) ||
+      this.isOfferQuantityInvalid(target, this.bank.theHerd) ||
+      !this.isSingleQuantityOnAnySide(offer, target)
     );
   }
 
-  private calculateValue([animal, count]: Offer): number {
-    return multiply(
-      ConvertAnimalName.toAnimalObject(animal).theValue,
-      count,
+  private isOfferQuantityInvalid(
+    offer: Offer[],
+    herd: Herd,
+  ): boolean {
+    return (
+      offer.filter(
+        ([animal, count]) => herd.getAnimalNumber(animal) >= count,
+      ).length !== offer.length
+    );
+  }
+
+  private isSingleQuantityOnAnySide(
+    offer: Offer[],
+    target: Offer[],
+  ): boolean {
+    return (
+      (offer.length === 1 && offer[0][1] === 1) ||
+      (target.length === 1 && target[0][1] === 1)
     );
   }
 
-  private adjustOffer(offer: Offer, target: Offer): void {
-    offer[1] -= 1;
+  private calculateValue(offer: Offer[]): number {
+    return sum(
+      offer.map(([animal, count]) =>
+        multiply(
+          ConvertAnimalName.toAnimalObject(animal).theValue,
```

Comment:
The Animal object with value is available on the players herd. Wouldn't it be easier to retrieve it from there, or perhaps send it together with name? Converting into object seems unnecessary.

Reply:
it can be time consuming, when each tupple in offer would be converted into another type by iteration through herd

Reply:
We have "one source of truth" which is our default configuration. Animals created in the herd are based on this configuration. I think we should not be creating other animals, especially that we wont be operating on them at all, they are created just for the value. Perhaps get it from configuration then? Please think about how to get it from config or the herd.

Reply:
Modified game and trade. now it is based on configuration